### ISSUE TITLE
Add definitions

### ISSUE THREAD

**Issue by sylwiavargas** [sylwiavargas/The-Code-Bending-Dictionary#6]
Effective definitions are at the core of the dictionary. Our dictionary contains definitions that could easily be classified as "explain it like I'm 5". We are serious about this.

## Before you contribute
1. Choose a word or words from *Words in need of definitions* section below. 
2. In the comments below, express that you want to write a definition and specify which word/s.
3. ✅ The easiest way to solicit different definitions is to ask someone who has learned these things recently. You can submit one PR per definition or a few definitions all at once — please contribute as much as as few as you wish!

🔥 **MAKE SURE THAT YOUR DEFINITIONS:**
- are beginners-friendly;
- do not contain confusing terms;
- are written in plain English;
- ideally, are funny and/or fluffy and/or kind and/or use an effective analogy;
- actually reflect what the thing is;
- align with our [Code of Conduct](https://github.com/sylwiavargas/The-Code-Bending-Dictionary/wiki/2.-Code-of-Conduct);

✚ you _can_ (but don't have to!) add a meme if there's one that connects to the term or your definition and a "read more" link if you know of any! If you are adding any, please read our guidelines for links and memes in this issue: #32 

## To Contribute

➡️ In order to contribute, please:
1. `git checkout dev` and `git checkout -b yourName`
2.  open the `seeds.rb` file and add new definition in the last section in an alphabetical order, following the pattern:

```ruby 
your_name_as_a_variable = User.create!(first_name: "your name or username", password: "just write abc123")

name_of_the_variable_def = Definition.create!(
  user: your_name_as_a_variable, 
  approved: true, 
  content: "here goes your content", 
  word: check_the_words_variable_in_the_words_section, 
  technology: check_the_technology_variable_in_the_technology_section,
  read_more_url: "some url", 
  read_more_at: "here write the name of the page", 
  meme_url: "some url", 
  meme_alt_text: "the description of the meme"
)
```

, for instance:

```ruby
eric = User.create!(first_name: "Eric", password: "abc123")


angry_squid_def = Definition.create!(
  user: foobar, 
  approved: true, 
  content: "'<%= %>': angry squid shoots ink or evaluates ruby code and prints", word: angry_squid, 
  read_more_url: "https://dev.to/sylwiavargas/which-erb-tag-should-i-use-53f1", 
  read_more_at: "Dev blog", 
  meme_url: "https://res.cloudinary.com/practicaldev/image/fetch/s--XEpd1g3r--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://dev-to-uploads.s3.amazonaws.com/i/qi9kko95vgpn0iknjaq0.png", 
  meme_alt_text:"a graphic erb cheatsheet for when to use erb tags or html. If code needs to be visible or return something you want to use the angry squid <%=", 
  technology: rails
)
```

3. Test your code by running `rails db:seed`
4. If your code does not generate new bugs, run `git add .`, `git commit -m "adds definition"`, `git push`
5. Submit a PR to `sylwiavargas:dev`

## Words in need of definitions

Each of the following words should have 2-4 different definitions:

**RUBY**:

- model,
- controller,
- params,
- strong params,
- ice-cream cone,
- angry squid,
- route,
- view,
- mvc,
- erb,
- self,
- oop,
- variable,
- method,
- object (Ruby),
- hash,
- array,
- instance,
- argument,
- parameter,
- instance variable,
- class variable,
- instance method,
- class method,
- module,
- attr_reader,
- attr_writer,
- attr_accessor,
- has_many,
- belongs_to,
- has_many: through:,
- single source of truth,
- schema,
- initialize,
- joiner,
- attribute,
- active record,
- rake,
- gem,
- gemfile,
- mass assignment,
- rollback,
- migration,
- database table,
- iterate,
- bundle,
- faker,
- tty prompt,
- seed,
- byebug,
- action,
- show action,
- index action,
- new action,
- edit action,
- create action,
- destroy action,
- update action,
- path helpers,
- html helpers,
- partials,
- layouts,
- validations,
- link_to,
- button_to,
- collection_select,
- redirect,
- render,
- named scope,
- guard clause,
- associations,
- flash hash,
- bcrypt,
- private method,

**General tech**:

- request,
- response,
- cli,
- single source of truth, 
- setter method,
- getter method,
- reader method,
- writer method,
- object oriented programming,
- functional programming,
- erd,
- Entity Relationship Diagram,
- domain model,
- one-to-many relationship,
- many-to-many relationship,
- json,
- api, 
- get request,
- put request,
- patch request,
- delete request,
- post request,
- web scraping,
- server,
- client,
- uri,
- url,
- rest,
- sessions,
- cookies,
- debugging
- crud,

**JS**:

- function,
- callback,
- hoisting,
- function expression,
- function invocation,
- function definition,
- debugger,
- console.log,
- let,
- const,
- var,
- variable,
- array,
- object,
- DOM,
- package,
- npm,
- node,
- document,
- querySelector,
- getElementById,
- querySelectorAll,
- localStorage,
- anonymous function,
- first-class functions,
- event listener,
- ecma,
- arrow functions,
- vanilla js,
- js
- fetch
- memoization

**React**:

- component
- props
- state
- jsx
- extend
- import
- export
- hooks
- react
- context
- src
- manifest
- lifecycle methods
- constructor
- render
- componentDidMount
- shouldComponentUpdate
- componentDidUpdate
- componentDidCatch
- componentWillUnmount
- useState
- useEffect
- useRef

**Comment by wlcreate** [comment]
Hi @sylwiavargas! I would like to add a definition for strong params :)

**Comment by wlcreate** [comment]
Hi @sylwiavargas! May I add a definition for flash hash?

**Comment by ChristyTropila** [comment]
> Hi @sylwiavargas. May I add definitions for CRUD, setter method, and getter method?

**Comment by bigfishh** [comment]
hey @ChristyTropila & @wlcreate! yes, go for it -- thank you!

**Comment by Shreya-L** [comment]
Hey @sylwiavargas, I added a definition for web scraping. Are we able to add multiple definitions in one pull request? Thanks!

**Comment by ChenyunZhang** [comment]
Hey  @sylwiavargas, May I add a definition for props, jsx, and extend?

**Comment by sylwiavargas** [comment]
@ChenyunZhang -> of course, go ahead and thank you! 
(Please make sure to work off the `dev` branch and then submit a PR to `dev`, not `main` ✨)

**Comment by nunnyr** [comment]
Hi @sylwiavargas 

May I add a definition for let, const, var ?

**Comment by sylwiavargas** [comment]
@nunnyr — yes, please! go ahead!
